src/game/Groups/GroupHandler.cpp

```diff
@@ -886,3 +886,44 @@ void WorldSession::HandleOptOutOfLootOpcode(WorldPacket& recv_data)
     if (unkn != 0)
         sLog.outError("CMSG_GROUP_PASS_ON_LOOT: activation not implemented!");
 }
+
+void WorldSession::HandleGroupSwapSubGroupOpcode(WorldPacket& recv_data)
+{
+    std::string playerName1, playerName2;
+
+    recv_data >> playerName1;
+    recv_data >> playerName2;
+
+    Group* group = GetPlayer()->GetGroup();
+    if (!group || !group->isRaidGroup())
+        return;
+
+    if (!group->IsLeader(GetPlayer()->GetObjectGuid()) &&
+        !group->IsAssistant(GetPlayer()->GetObjectGuid()))
+        return;
+
+    auto getGuid = [&group](std::string const& playerName)
+    {
+        if (Player* player = sObjectMgr.GetPlayer(playerName.c_str()))
+            return player->GetObjectGuid();
+        else
+        {
+            if (ObjectGuid guid = sObjectMgr.GetPlayerGuidByName(playerName))
+                return guid;
+            else
+                return ObjectGuid();
+        }
+    };
+
+    ObjectGuid guid1 = getGuid(playerName1);
```

Comment:
Exiting the function as soon as either name query fails may be useful. One full name search can be avoided if first name is unknown.
```
if (!guid)
    return;
```